lbrynet/core/HTTPBlobDownloader.py

```diff
@@ -0,0 +1,81 @@
+from random import choice
+import logging
+
+from twisted.internet import defer
+import treq
+from twisted.internet.task import LoopingCall
+
+log = logging.getLogger(__name__)
+
+
+class HTTPBlobDownloader(object):
+    def __init__(self, blob_manager, blob_hashes=None, servers=None, client=None):
+        self.blob_manager = blob_manager
+        self.servers = servers or []
+        self.client = client or treq
+        self.blob_hashes = blob_hashes or []
+        self.looping_call = LoopingCall(self._download_next_blob_hash_for_file)
+        self.failures = 0
+        self.max_failures = 3
+        self.interval = 1
+
+    @property
+    def running(self):
+        return self.looping_call.running
+
+    def start(self):
+        if not self.running and self.blob_hashes and self.servers:
+            return self.looping_call.start(self.interval, now=True)
+
+    def stop(self):
+        if self.running:
+            self.blob_hashes = []
+            return self.looping_call.stop()
+
+    @defer.inlineCallbacks
+    def _download_next_blob_hash_for_file(self):
+        for blob_hash in self.blob_hashes:
+            blob = yield self.blob_manager.get_blob(blob_hash)
+            if not blob.get_is_verified():
+                self.download_blob(blob)
+                defer.returnValue(None)
+        self.stop()
+
+    def download_blob(self, blob):
+        d = self._download_blob(blob)
+        d.addCallback(self._on_completed_blob)
+        d.addErrback(self._on_failed)
```

Comment:
I think we're moving towards using `@defer.inlineCallbacks` more because it's easier to read / reason about and will make a some-day port to asyncio easier. And in this particular case it will have the added advantage of reducing the class surface area by moving the two methods below directly inline into this method.